### PR TITLE
test: migrate part of Notification tests

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

test: migrate part of Notification tests

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/notification/__tests__/config.test.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ describe('notification.config', () => {
       act(() => {
         notification.open({
           message: 'Notification message',
-          key: i,
+          key: i.toString(),
```
Comment (zombieJ) [inline components/notification/__tests__/config.test.ts:23]: as 一下，这个要保留，以防止 key 类型变化导致的问题。

File: components/notification/__tests__/hooks.test.tsx [hunk 2]
```diff
@@ -72,36 +75,37 @@ describe('notification.hooks', () => {
...
-      const [, forceUpdate] = React.useState({});
+    const Demo: React.FC = () => {
+      const [, forceUpdate] = React.useState([]);
       const [api] = notification.useNotification();
-
       React.useEffect(() => {
-        count += 1;
-        expect(count).toEqual(1);
-        forceUpdate();
+        count++;
```
Comment (zombieJ) [inline components/notification/__tests__/hooks.test.tsx:101]: 奇怪，++ eslint 应该报错才对。。。 `+= 1` 才是对的。

Comment (li-jia-nan) [reply components/notification/__tests__/hooks.test.tsx:101]: > 奇怪，++ eslint 应该报错才对。。。 `+= 1` 才是对的。

那我改回去

File: components/notification/__tests__/config.test.ts (hunk 1 above)
Comment (li-jia-nan) [reply components/notification/__tests__/config.test.ts:23]: > as 一下，这个要保留，以防止 key 类型变化导致的问题。

我改一下

File: components/notification/__tests__/hooks.test.tsx [hunk 3]
```diff
@@ -72,36 +75,37 @@ describe('notification.hooks', () => {
...
+        expect(count).not.toBe(0);
```
Comment (zombieJ) [inline components/notification/__tests__/hooks.test.tsx:102]: 使用 test util 里的 pureRender 可以变成非 strictMode，否则这个测试就不等价了

Comment (li-jia-nan) [reply components/notification/__tests__/hooks.test.tsx:102]: > 使用 test util 里的 pureRender 可以变成非 strictMode，否则这个测试就不等价了

我改一下

File: components/notification/__tests__/hooks.test.tsx [hunk 4]
```diff
@@ -72,36 +75,37 @@ describe('notification.hooks', () => {
...
-      const [, forceUpdate] = React.useState({});
+    const Demo: React.FC = () => {
+      const [, forceUpdate] = React.useState([]);
       const [api] = notification.useNotification();
-
       React.useEffect(() => {
         count += 1;
-        expect(count).toEqual(1);
-        forceUpdate();
+        expect(count).not.toBe(1);
```
Comment (zombieJ) [inline components/notification/__tests__/hooks.test.tsx:102]: 不等价的

**Pull request merged**